language/tools/move-package/src/resolution/dependency_graph.rs

```diff
@@ -336,6 +353,78 @@ impl DependencyGraph {
         Ok(())
     }
 
+    /// Add the graph in `other` to `self` consuming it in the process.  Assumes the root of `other`
+    /// is the only shared node between the two, and fails if this is not the case.  Labels packages
+    /// coming from `other` as being resolved by `resolver`.
+    ///
+    /// It is an error to attempt to merge into `self` after its `always_deps` (the set of packages
+    /// that are always transitive dependencies of its root, regardless of mode) has been
+    /// calculated.  This usually happens when the graph is created, so this function is intended
+    /// primarily for internal use, but is exposed for testing.
+    pub fn merge(&mut self, other: DependencyGraph, resolver: Symbol) -> Result<()> {
```

Comment:
Super minor nit (that you can ignore if you disagree): but `other` here (and the use of `other_<name>` below) while a good name doesn't  feel quite as clear as it could be. What would you think of something like `external`?

Reply:
Is the inspiration here that the `other` graph is coming from external resolution?  That's definitely true for how `merge` is currently used, but I don't think it's intrinsic to the operation.  I won't defend `other` as a great name though -- what about ... `sub_graph`, or ... `extension`?  This recognises the fact that `other` needs to attach into a point in `self`.